go/connector/run.go

```diff
@@ -145,6 +146,16 @@ func Run(
 		"--network", networkName,
 	}
 
+	if containerName != "" {
+		// Replace non-alphanum chars.
+		// Container names must match [a-zA-Z0-9][a-zA-Z0-9_.-]+.
+		// FIXME handle names prefixed with [_.-+/]+.
+		containerName = strings.NewReplacer("+", "_", "/", "_").Replace(containerName)
```

Comment:
I'm realizing that there could be name collisions here. You could have two different Flow tasks, `foo/bar/baz` and `foo/bar_baz`, and both of those would result in the same `containerName`.

I'm not sure what we do about that. One option would be to base64 the whole provided name, but that might make it difficult to correlate the metrics later. Although, now that I look at the [cAdvisor docs](https://github.com/google/cadvisor/blob/master/docs/runtime_options.md#container-labels), I see that it's able to pass through labels! So maybe we can just base64 the provided name in order to get a valid container name, and add the original name as a label to allow the metrics to be correlated with the shard.

Reply:
Nice, labeling is much more elegant! We can just keep docker assigned IDs and use labels for metrics tagging. 
This setup won't restrict duplicate containers from orchestrator misbehavior though. Tags will look something like, 
```
df6abb0ec43a    keen_ellis      shard=materialize/examples/stats/00000000-00000000,FLOW_RUNTIME_PROTOCOL=materialize,FLOW_TCP_PORT=2222,command=transactions
2b791ba97745    kind_shtern     FLOW_RUNTIME_PROTOCOL=materialize,FLOW_TCP_PORT=2222,command=transactions,shard=materialize/test/hello-world/ssh-postgresl/00000000-00000000
```